project/plugins.sbt

```diff
@@ -1,6 +1,3 @@
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "3.9.7")
-addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
-addSbtPlugin("com.dwijnand" % "sbt-dynver" % "4.1.1")
```

Comment:
This repository implements the steps described in the [Library Author Guide](https://docs.scala-lang.org/overviews/contributors/index.html#setup-continuous-publication) where we also explain how to publish step by step before showing the "one-command" approach of `ci-release`. In this case we explicitly use these dependencies.

Even if they are transitively depended on by `ci-release`, is it a good a idea to explicitly depend on them, in case `ci-release` won't use them someday?

Reply:
Since we configure settings from sbt-sonatype in our build.sbt, I would keep that plugin dependency. Although, maybe if someday sbt-ci-release stops using sbt-sonatype, then our configuration will have to be migrated to something else anyway…?

Similar reasoning can be made for sbt-dynver and sbt-pgp: if someday sbt-ci-release stops using them, it will provide another mechanism and our dependency on sbt-dynver or sbt-pgp might even create conflicts?

So, I guess we can remove these dependencies.

Reply:
You probably want to get compile errors for sbt-sonatype references if sbt-ci-release removes that dependency in the future (which is highly unlikely, I’d probably just create a new release plugin instead). 

I think it’s nice to remove these implicit dependencies because 1. it reduces boilerplate when creating a new build and 2. it avoids Scala steward noise because you only update sbt-ci-release (we try to release new versions as soon as upstream dependencies become available). 

I think it’s totally fine to customize sonatype/dynver settings even if they’re brought in transitively. I don’t consider those plugins as implementation details, they are the API of sbt-ci-release.